### PR TITLE
feat(mongodb): deprecate createCollection options

### PR DESCRIPTION
**Pull request by emmanuelgautier**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/mongodb/node-mongodb-native/commit/4cc6bcc890f91d723865f3e7e6e58f4aa3985f89
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/mongodb/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for MongoDB 3.5
+// Type definitions for MongoDB 3.6.2
```
Comment (peterblazejewicz) [inline types/mongodb/index.d.ts:1]:
```suggestion
// Type definitions for MongoDB 3.6
```

Comment (emmanuelgautier) [reply types/mongodb/index.d.ts]: Thanks @peterblazejewicz

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## mongodb ([<kbd>unpkg</kbd>](https://unpkg.com/browse/mongodb@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'mongodb'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 9e8095646654ca5c22538065274629997a8e52cc
</p>

File: types/mongodb/index.d.ts [hunk 2]
```diff
@@ -864,8 +865,15 @@ export interface CollectionCreateOptions extends CommonOptions {
...
+    /**
+     * @deprecated
+     * @see https://jira.mongodb.org/browse/NODE-2746
+     */
     strict?: boolean;
     capped?: boolean;
+    /**
+     * @deprecated
+     */
     autoIndexId?: boolean;
```
Comment (LinusU) [inline types/mongodb/index.d.ts:877]:
```suggestion
```

This option was removed in MongoDB 3.4, and the typings are for 3.6

ref: https://stackoverflow.com/a/41783524/148072

**Review by LinusU — APPROVED** [review]

File: types/mongodb/index.d.ts (hunk 2 above)
Comment (emmanuelgautier) [reply types/mongodb/index.d.ts:877]: We should maybe keep this property to let the developers using this one know that is deprecated before removing it ?

Comment (orta) [reply types/mongodb/index.d.ts:877]: Giving a chance to make this discussion

**Comment by elibarzilay** [comment]
@LinusU, @emmanuelgautier: any resolution re the above discussion?

**Review by HitkoDev — APPROVED** [review]

**Comment by emmanuelgautier** [comment]
Ready to merge

**Pull request merged**